draft/src/api.rs

```diff
@@ -3,16 +3,79 @@ use std::fmt;
 #[derive(Debug, Clone, Hash, Eq, PartialEq)]
 pub struct Card {
     pub name: String,
+    pub mana_cost: ManaCost,
 }
 
 impl Card {
-    pub fn new(name: &str) -> Self {
+    pub fn new(name: &str, mana_cost: ManaCost) -> Self {
         Self {
-           name: name.to_string(),
+            name: name.to_string(),
+            mana_cost,
         }
     }
 }
 
+#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq)]
+pub enum BasicMana{
+    Red,
+    Green,
+    Blue,
+    Black,
+    White,
+    // Specifically requests colorless mana.
+    Colorless
+}
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub enum Mana{
+    XCost(String),
+    Phyrexian(BasicMana),
+    Split(BasicMana, BasicMana),
+    Standard(BasicMana),
+    NumGeneric(i16),
+    // paid with either 1 of the given basic or 2 generic mana
+    SplitGeneric(BasicMana)
```

Comment:
Not sure if it matters or not, but Snow Mana isn't currently included here, but it is a distinct mana type.

Nitpick: the official term for "Split" mana is "Hybrid" mana.

Reply:
I don't think we should get too deep into trying to fully capture every detail of magic mana right now, especially since the first major goal is to get a working ai for new cube, which generally avoids weird mana costs. I think Hybrid and X are the only nonstandard ones used in new cube.

Reply:
Okay, I'll leave this alone for now.